test/boulder-tools/tag_and_upload.sh

```diff
@@ -10,18 +10,18 @@ GO_VERSIONS=( "1.17" "1.17.5" )
 echo "Please login to allow push to DockerHub"
 docker login
 
+# create a docker buildx node for cross-compilation.
+docker buildx create --use --name=cross
+
+# on EXIT, delete the docker buildx node if it exists.
+trap "if [ $(docker buildx ls | grep 'cross' | wc -l) != 0 ]; then docker buildx rm cross; fi" EXIT
```

Comment:
```suggestion
trap "if docker buildx ls | grep -q 'cross'; then docker buildx rm cross; fi" EXIT
```

`if` runs a command (commonly `/bin/[`, or its built-in implementation), and executes the body if that command returns 0.

Also, `grep` returns 0 if it finds its pattern anywhere.

Also, why delete the buildx node at exit rather than right before we create it? For that matter, why not reuse it if it already exists?

Reply:
Thanks for picking at this, after re-reading the docs it turns out that I implemented this create/delete pattern wrong the first time. In-fact, creating the node if it doesn't already exist and then keeping it around improves our docker image build times pretty substantially as it allows the builder to use the local cache for various steps.

You can ignore the WARN in this screenshot, I was doing no push builds to try this out:
<img width="880" alt="image" src="https://user-images.githubusercontent.com/37676056/146455031-584ce8f2-ec06-4eed-8c9d-5d2998145aab.png">

This shaves a full 400 seconds off of our builds.

Thanks!